docs/src/main/tut/generate-sources-from-idl.md

```diff
@@ -0,0 +1,301 @@
+---
+layout: docs
+title: Generating sources from IDL
+permalink: /generate-sources-from-idl
+---
+
+# Generate sources from IDL
+
+Since `Protobuf` and `Avro` are both a language-neutral, platform-neutral way of serializing structured data, our aim with [mu] is to create Scala definitions from these IDL files. 
+Before going into implementation details let's see how to create the Scala files with our definitions.
+
+Currently, `Avro` is supported in both `.avpr` (JSON) and `.avdl` (Avro IDL) formats, along with `Protobuf`.
+The plugin's implementation basically wraps the [avrohugger] library for `Avro` 
+and uses [skeuomorph] for `Protobuf` adding some mu-specific extensions.
+## Plugin Installation
+
+Add the following line to _project/plugins.sbt_:
+
+[comment]: # (Start Replace)
+
+```scala
+addSbtPlugin("io.higherkindness" % "sbt-mu-idlgen" % "0.18.3")
+```
+
+[comment]: # (End Replace)
+
+Note that the plugin is only available for Scala 2.12.
+
+
+### Plugin Settings
+
+For generating Scala definitions we use `srcGen`. The easiest way to use the plugin is integrating the source generation in your compile process by adding this import to your `build.sbt` file:
+
+```scala
+import higherkindness.mu.rpc.idlgen.IdlGenPlugin.autoImport._
+``` 
+and the setting,   
+                                                                                
+```scala  
+idlType := proto  
+sourceGenerators in Compile += (srcGen in Compile).taskValue
+```
+
+Otherwise, you can run the following sbt task (with the settings above in your `build.sbt`):
+```sbtshell
+sbt
+project foo
```

Comment:
I'd omit the `foo` project

Reply:
I think it’s easier to understand for first-time-users that use multi-module project. Maybe it's better to keep unimodule project way, I'm not sure.